samples/net/mqtt_publisher/src/main.c

```diff
@@ -420,57 +393,32 @@ static int network_setup(struct net_context **net_ctx, const char *local_addr,
 	printk("\n");
 #endif
 
-	rc = set_addr(&local_sock, local_addr, 0);
-	if (rc) {
-		printk("set_addr (local) error\n");
+#ifdef CONFIG_NET_IPV6
+	rc = net_addr_pton(AF_INET6, CONFIG_NET_APP_MY_IPV6_ADDR, &addr);
+	if (rc < 0) {
+		NET_ERR("Invalid address: %s", CONFIG_NET_APP_MY_IPV6_ADDR);
 		return rc;
 	}
 
-#ifdef CONFIG_NET_IPV6
 	p = net_if_ipv6_addr_add(net_if_get_default(),
-				 &net_sin6(&local_sock)->sin6_addr,
-				 NET_ADDR_MANUAL, 0);
-#else
-	p = net_if_ipv4_addr_add(net_if_get_default(),
-				 &net_sin(&local_sock)->sin_addr,
-				 NET_ADDR_MANUAL, 0);
-#endif
-
+				&addr, NET_ADDR_MANUAL, 0);
 	if (!p) {
 		return -EINVAL;
 	}
-
-	rc = net_context_get(family, SOCK_STREAM, IPPROTO_TCP, net_ctx);
-	if (rc) {
-		printk("net_context_get error\n");
+#else
+	rc = net_addr_pton(AF_INET6, CONFIG_NET_APP_MY_IPV4_ADDR, &addr);
+	if (rc < 0) {
+		NET_ERR("Invalid address: %s", CONFIG_NET_APP_MY_IPV4_ADDR);
 		return rc;
 	}
-
-	rc = net_context_bind(*net_ctx, &local_sock, addr_len);
-	if (rc) {
-		printk("net_context_bind error\n");
-		goto lb_exit;
-	}
-
-	rc = set_addr(&server_sock, server_addr, server_port);
-	if (rc) {
-		printk("set_addr (server) error\n");
-		goto lb_exit;
-	}
-
-	rc = net_context_connect(*net_ctx, &server_sock, addr_len, NULL,
-				 APP_SLEEP_MSECS, NULL);
-	if (rc) {
-		printk("net_context_connect error\n");
-		goto lb_exit;
+	p = net_if_ipv4_addr_add(net_if_get_default(),
```

Comment:
If you call net_app_init(), then that function will make sure that the IP addresses are set to the network interface and you can remove this and the corresponding ipv6 call from the app.